packages/ui/src/models/visualization/flows/nodes/mappers/parallel-processor-base-node-mapper.ts

```diff
@@ -0,0 +1,73 @@
+import { ProcessorDefinition } from '@kaoto/camel-catalog/types';
+import { NodeIconResolver, NodeIconType } from '../../../../../utils/node-icon-resolver';
+import { IVisualizationNode } from '../../../base-visual-entity';
+import { createVisualizationNode } from '../../../visualization-node';
+import { CamelRouteVisualEntityData, ICamelElementLookupResult } from '../../support/camel-component-types';
+import { BaseNodeMapper } from './base-node-mapper';
+import { CamelComponentSchemaService } from '../../support/camel-component-schema.service';
+import { getValue } from '../../../../../utils';
+
+export abstract class ParallelProcessorBaseNodeMapper extends BaseNodeMapper {
+  abstract getProcessorName(): keyof ProcessorDefinition;
+
+  getVizNodeFromProcessor(
+    path: string,
+    _componentLookup: ICamelElementLookupResult,
+    entityDefinition: unknown,
+  ): IVisualizationNode {
+    const processorName = this.getProcessorName();
+
+    const data: CamelRouteVisualEntityData = {
+      path,
+      icon: NodeIconResolver.getIcon(processorName, NodeIconType.EIP),
+      processorName,
+      isGroup: true,
+    };
+
+    const vizNode = createVisualizationNode(processorName, data);
+    const children = this.getChildrenFromBranch(`${path}.steps`, entityDefinition);
+    children.forEach((child) => {
+      vizNode.addChild(child);
+    });
+
+    return vizNode;
+  }
+
+  getChildrenFromBranch(path: string, entityDefinition: unknown): IVisualizationNode[] {
+    const stepsList = getValue(entityDefinition, path, []) as ProcessorDefinition[];
+
+    return stepsList.reduce((accStepsNodes, step, index) => {
+      const singlePropertyName = Object.keys(step)[0];
+      const childPath = `${path}.${index}.${singlePropertyName}`;
+      const childComponentLookup = CamelComponentSchemaService.getCamelComponentLookup(
+        childPath,
+        getValue(step, singlePropertyName),
+      );
+
+      const vizNode = this.rootNodeMapper.getVizNodeFromProcessor(childPath, childComponentLookup, entityDefinition);
+      this.addNodeInteraction(vizNode, childComponentLookup.processorName as keyof ProcessorDefinition);
+      accStepsNodes.push(vizNode);
+      return accStepsNodes;
+    }, [] as IVisualizationNode[]);
+  }
```

Comment:
This seems very similar to the [base-node-mapper](https://github.com/KaotoIO/kaoto/blob/972d250151e6c43a8349d9b346c44b7b67589cae/packages/ui/src/models/visualization/flows/nodes/mappers/base-node-mapper.ts#L73-L95) mapper.

Wouldn't be easier to remove this, and simply iterate over the children and remove the previous and next steps instead?

```typescript
  getVizNodeFromProcessor(
    path: string,
    _componentLookup: ICamelElementLookupResult,
    entityDefinition: unknown,
  ): IVisualizationNode {
    const processorName = this.getProcessorName();

    const data: CamelRouteVisualEntityData = {
      path,
      icon: NodeIconResolver.getIcon(processorName, NodeIconType.EIP),
      processorName,
      isGroup: true,
    };

    const vizNode = createVisualizationNode(processorName, data);
    const children = this.getChildrenFromBranch(`${path}.steps`, entityDefinition);
    children.forEach((child) => {
      vizNode.addChild(child);
      vizNode.setPreviousNode(undefined);
      vizNode.setNextNode(undefined);
    });

    return vizNode;
  }
```

Of course, this would need to update the IVizNode interface and implementation:
```
  setPreviousNode(previousNode: IVisualizationNode | undefined) {
    this.previousNode = previousNode;
  }

  setNextNode(node: IVisualizationNode | undefined) {
    this.nextNode = node;
  }
```